js/main_page.js

```diff
@@ -1,55 +1,41 @@
 /* 
-
 Javascript for MasterMind Gems
-
 Front/Main page loading
-
 */
 
-function showPage(id) {
-	// Internal page navigation only
-	let element = document.getElementById(id);
-	element.style.display="grid";
-	window.location = "#" + id;
-	// element.style.animation-play-state="paused";
-	// document.getElementById(id).style.display="grid";
-}
-
-
-function hidePage(id) {
-	let element = document.getElementById(id);
-	// if (element.classList.contains("onstart-load")) {
-	// 	element.classList.remove("onstart-load");
-	// }
-	element.style.display="none";
-
-	// let elements = document.getElementById(id), element_children = elements.getElementsByName("A");
-	// element_children.style
-
-	// let div_parent = document.getElementById(row_no),
- //            div_children = div_parent.getElementsByTagName("DIV");
-
-}
-
-
-function hideAndShow(hideId, showId) {
-	let current_pg = document.getElementById(hideId);
-
-	if (current_pg.classList.contains("onstart-load")) {
-		current_pg.classList.remove("onstart-load");
-	}
-
-	current_pg.className += " unload";
-
-	window.setTimeout( () => {
-		hidePage(hideId);
-		showPage(showId);
-
-		// remember to remove the unload class from the previous element
-		if (current_pg.classList.contains("unload")) {
-			current_pg.classList.remove("unload");
-		}
-
-	}, 2000);
-	
-}
\ No newline at end of file
+// Internal page navigation only
+const showPage = pageId => {
+  const element = document.getElementById(pageId);
+  element.style.display = "grid";
+  window.location = "#" + pageId;
+};
+
+const hidePage = pageId => {
+  const element = document.getElementById(pageId);
+  element.style.display = "none";
+};
+
+const animatePageHiding = pageId => {
+  const current_pg = document.getElementById(pageId);
+  if (current_pg.classList.contains("onstart-load")) {
+    current_pg.classList.remove("onstart-load");
+  }
+  current_pg.className += " unload";
+};
+
+const resetPageStyle = () => {
+  const current_pg = document.getElementById(pageId);
+  if (current_pg.classList.contains("unload")) {
+    current_pg.classList.remove("unload");
+  }
+};
+
+const hideAndShow = (hideId, showId) => {
+  animatePageHiding(hideId);
+
+  window.setTimeout(() => {
+    hidePage(hideId);
+    showPage(showId);
+    resetPageStyle();
+  }, 2000);
+};

```

Comment:
* I noticed that arrow functions are being used in many places in the project. It's better to keep the coding style consistent. Using arrow functions now is recommended as it has many benefits and it's also more simple to read.

* It's recommended to use `const` all the time unless our object will be assigned a value later in our code. Using `const` still means that we can mutate the properties of the object, but it will not allow us to re-assign a new value to the object itself. For example: `current_pg`. Assigning `const` to `current_pg` means that we can't do something like that in our code: `current_pg = new_object` but it means that we can mutate and change the properties of `current_pg` itself. For example, we can change the property `className` of `current_pg`.

* It's recommended to keep the case style consistent. `camelCase` is more common in the code base, so it is better if we change `current_pg` for example to be `currentPg`.

* Choosing descriptive names for the function arguments is recommended. For example, instead of having an argument called `id` it's better to rename it to something like `pageId`.

* Encapsulating related logic in one function can be recommended some times for readability. For example, we can encapsulate all the logic needed for animating the page hiding in one function `animatePageHiding`. Also the same applies for `resetPageStyle`.